testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/jms/ClusteredMessagingTestCase.java

```diff
@@ -46,145 +46,118 @@
-            jmsOperations.createJmsTopic(jmsTopicName, "java:jboss/exported/" + jmsTopicLookup);
-        }
-        try (ModelControllerClient client = createClient2()) {
+        // Create JMS Queue and Topic (the ServerSetupTask won't do the trick since there is no deployment in this test case...)
+        Arrays.stream(new ModelControllerClient[] { createClient1(), createClient2() }).forEach(client -> {
             JMSOperations jmsOperations = JMSOperationsProvider.getInstance(client);
             jmsOperations.createJmsQueue(jmsQueueName, "java:jboss/exported/" + jmsQueueLookup);
             jmsOperations.createJmsTopic(jmsTopicName, "java:jboss/exported/" + jmsTopicLookup);
-        }
+        });
     }
 
     @Override
     public void afterTestMethod() throws Exception {
         super.afterTestMethod();
 
-        try (ModelControllerClient client = createClient1()) {
+        // Remove JMS Queue and Topic
+        Arrays.stream(new ModelControllerClient[] { createClient1(), createClient2() }).forEach(client -> {
             JMSOperations jmsOperations = JMSOperationsProvider.getInstance(client);
             jmsOperations.removeJmsQueue(jmsQueueName);
             jmsOperations.removeJmsTopic(jmsTopicName);
-        }
-        try (ModelControllerClient client = createClient2()) {
-            JMSOperations jmsOperations = JMSOperationsProvider.getInstance(client);
-            jmsOperations.removeJmsQueue(jmsQueueName);
-            jmsOperations.removeJmsTopic(jmsTopicName);
-        }
+        });
     }
 
     @Test
     public void testClusteredQueue() throws Exception {
-        InitialContext contextFromServer0 = createJNDIContextFromServer0();
-        InitialContext contextFromServer1 = createJNDIContextFromServer1();
+        InitialContext contextFromServer1 = createJNDIContext(NODE_1);
+        InitialContext contextFromServer2 = createJNDIContext(NODE_2);
 
         String text = UUID.randomUUID().toString();
 
-        // WIP test if the problem is that the view is not yet propagated
-        Thread.sleep(GRACE_TIME_TO_MEMBERSHIP_CHANGE);
-
-        // send to the queue on server 0
-        sendMessage(contextFromServer0, jmsQueueLookup, text);
-        // receive it from the queue on server 1
-        receiveMessage(contextFromServer1, jmsQueueLookup, text);
+        // send to the queue on server 1
+        sendMessage(contextFromServer1, jmsQueueLookup, text);
+        // receive it from the queue on server 2
+        receiveMessage(contextFromServer2, jmsQueueLookup, text);
 
         String anotherText = UUID.randomUUID().toString();
-        // send to the queue on server 1
-        sendMessage(contextFromServer1, jmsQueueLookup, anotherText);
-        // receive it from the queue on server 0
-        receiveMessage(contextFromServer0, jmsQueueLookup, anotherText);
+        // send to the queue on server 2
+        sendMessage(contextFromServer2, jmsQueueLookup, anotherText);
```

Comment:
You are using the same destination as for creating the consumer. It would be nicer to avoid a 2nd JNDI lookup.

Reply:
OK; I extracted the destination lookups and reused.